modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/Opener.js

```diff
@@ -15,14 +15,36 @@
 import ClayIcon from '@clayui/icon';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import React, {useContext} from 'react';
+import React, {useContext, useEffect, useState} from 'react';
 
 import MiniCartContext from './MiniCartContext';
 
 function Opener({openCart}) {
-	const {cartState, spritemap} = useContext(MiniCartContext),
-		{cartItems = []} = cartState,
-		numberOfItems = cartItems.length > 0 ? cartItems.length : 0;
+	const {cartState, displayTotalItemsQuantity, spritemap} = useContext(
+		MiniCartContext
+	);
+
+	const {cartItems, itemsQuantity: initialItemsQuantity, summary} = cartState,
+		[numberOfItems, setNumberOfItems] = useState(0);
+
+	useEffect(() => {
+		setNumberOfItems(initialItemsQuantity);
+
+		return () => {};
```

Comment:
I don't think you need to return a function here if it doesn't do anything — React will call it, it will do nothing, but React would have been just fine if you had returned nothing at all.

Reply:
I think I've been overly zealous with the "remember to always clean effects up"...